### PR TITLE
Update resolver documentation

### PR DESCRIPTION
**Pull request by fsat**
- Update the resolver documentation with supported URI schemes.
- Describe the bundle resolution process.
- Document additional resolvers: S3, stdin, and offline.
- Update the custom resolver example with supported URI scheme.

### CODE REVIEW

**Comment by fsat** [comment]
Needs cherry-picking to `2.1` and `2.0` branches.

**Review by markusjura — APPROVED** [review]
Overall, looking good. Created a new commit to this PR with some grammar and style changes. Please review these and squash the commits before merging.

File: src/main/play-doc/operation/DeployingBundlesOps.md [hunk 1]
```diff
@@ -80,21 +80,21 @@ In the example above the bundle shorthand expression `reactive-maps-backend-regi
...
 
-The list of possible URI scheme from input is then compared to the list of URI schemes supported by each resolver in the selected chain. Only resolvers which are able to accept URI scheme from input is used during bundle resolution process.
+The list of possible URI schemes is then compared to the list of URI schemes supported by each resolver in the selected chain. Only resolvers which can accept the selected URI schemes are used during the bundle resolution process.
 
-The input is then passed into each remaining resolver in the chain until a successful resolution can be found. The result of a successful resolution is the bundle artefact which will then be cached. If none of the remaining resolver in the chain is able to resolve the input, an error is raised.
+The input is then passed into each remaining resolver in the chain until a successful resolution can be found. The result of a successful resolution is the bundle artifact which will then be cached. If none of the remaining resolvers in the chain can resolve the input, an error is raised.
```
Comment (markusjura) [inline src/main/play-doc/operation/DeployingBundlesOps.md]: Artifact with an `i` is american english which is what we use in our documentation.

File: src/main/play-doc/operation/DeployingBundlesOps.md [hunk 2]
```diff
@@ -148,9 +148,13 @@ The CLI comes with a number of built-in resolvers that come into play when `cond
 
 ### URI resolver
 
-Located at `conductr_cli.resolvers.uri_resolver` Python package.
+Located at the [conductr_cli.resolvers.uri_resolver](https://github.com/typesafehub/conductr-cli/blob/master/conductr_cli/resolvers/uri_resolver.py) Python package.
```
Comment (markusjura) [inline src/main/play-doc/operation/DeployingBundlesOps.md]: Now we link to the respective Python package source code.

**Comment by fsat** [comment]
#397 takes care of `2.1`

**Comment by fsat** [comment]
#398 takes care of `2.0`

**Pull request merged**